src/main/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseService.java

```diff
@@ -40,7 +41,22 @@ public CourtCaseEntity createCase(String courtCode, String caseNo, CourtCaseEnti
                         },
                         () -> trackCreateEvents(updatedCase));
 
-        return courtCaseRepository.save(updatedCase);
+        return Mono.just(updatedCase)
+            .map(courtCaseRepository::save)
+            .doAfterTerminate(() -> updateStatusForCrn(updatedCase.getCrn(), updatedCase.getProbationStatus(), updatedCase.getCaseNo()));
```

Comment:
Reading of the API is that this should happen after the Mono has completed. However tests so far are unclear when. Should we consider a log debug statement on the save operation ?

Reply:
Yeah that would be good, I'd be tempted to create a telemetry event for it in fact 🤔  We'll get telemetry on the updates for each case though but might be useful to tie that back to a single probation status change.. but then again that's also covered by existing telemetry. No, maybe I've talked myself out of it, debug log is probably right! 😄

Reply:
log statement done.